Prova di consegna - Saldana Moreno Stefano Enanias

Wayde07 opened the pull request:
Non sono riuscito a capire come formattare bene il testo all'interno della lista dei prodotti, per il resto credo finalmente di aver capito qualcosa :)

Wayde07 committed:
Prova di consegna

Non sono riuscito a capire come formattare bene il testo all'interno della lista dei prodotti, per il resto credo finalmente di aver capito qualcosa :)

danielebarbaro commented on tempCodeRunnerFile.mjs:
```diff
@@ -0,0 +1,2 @@
+
```
cos'è questo file?

Wayde07 replied:
È quello che mi crea VA in automatico per non so che cosa, prossima volta mi ricorderò di non pusharlo, grazie mille

danielebarbaro commented on core-cart.mjs:
```diff
@@ -5,6 +5,75 @@ import * as os from "os";
+}
+
+const printReceipt = (testo, contatore) => {
+    fs.writeFile(`./receipt/scontrino${contatore}.txt`, testo, err => {
```
Leggi bene le richieste dentro il file readme manca qualcosa.

Wayde07 replied:
Si purtroppo non sono riuscito bene a implementare tutte le richieste non avendole assimilate molto bene :(

danielebarbaro replied:
la cartella di atterraggio è `receipts` non `receipt` come menzionato nel README

Il nome del file doveva essere : `$UserUUID_receipt_$DATAORAGENERAZIONE.txt` dove `$UserUUID` è lo UUID dell'utente e $DATAORAGENERAZIONE è la data.

invece di chiamare: `core.printReceipt(scontrino, contatore);` (in cart.mjs) avresti dovuto chiamare `core.printReceipt(scontrino, cartUserUUID);`

Se riesci prova a modificare questo metodo per avere il nome corretto.

danielebarbaro commented on core-cart.mjs:
```diff
@@ -5,6 +5,75 @@ import * as os from "os";
+
+const formatProductName = (product) => product;
+
+const AlterType = (products, type) => products;
```
? cosa fa questo metodo

Wayde07 replied:
Li ho lasciati vuoti, dovrei toglierli tutti quelli che non ho usato, mi sono dimenticato

danielebarbaro commented on core-cart.mjs:
```diff
@@ -5,6 +5,75 @@ import * as os from "os";
+    return listaProdotti;
+}
+
+const formatProductName = (product) => product;
```
? cosa fa questo metodo

danielebarbaro commented on core-cart.mjs:
```diff
@@ -5,6 +5,75 @@ import * as os from "os";
+
+const getProductByCart = (products) => products;
+
+const getUserDiscount = () => 0;
```
? cosa fa questo metodo

danielebarbaro commented on core-cart.mjs:
```diff
@@ -5,6 +5,75 @@ import * as os from "os";
+    
+}
+
+const getProductByCart = (products) => products;
```
? cosa fa questo metodo

danielebarbaro commented on core-cart.mjs:
```diff
@@ -5,6 +5,75 @@ import * as os from "os";
+
+const getUserDiscount = () => 0;
+
+const receiptFileName = (uuid, date) => '';
```
? cosa fa questo metodo

danielebarbaro commented on core-cart.mjs:
```diff
@@ -5,6 +5,75 @@ import * as os from "os";
+    let listaProdotti = ``;
+    lista.forEach(codice => {        
+        let prodotto = products.find(product => product.ean === codice);
+        listaProdotti += prodotto.ean 
```
manca un `\n` prima di `prodotto.ean` che ti manda a capo le righe.

Wayde07 replied:
Ah ecco, grazie mille

Wayde07 committed:
Correzioni parziali

Ho corretto meglio l'impaginazione dello scontrino, manca sempre la formattazione testo dei prodotti ma ci sto lavorando

Wayde07 committed:
Scontrini

Scontrini della soluzione precedente

Wayde07 committed:
Formattazione prodotti corretta

Wayde07 committed:
Formattazione corretta

Mi ero dimenticato delle parole composta, sistemato anche quelllo :)

Wayde07 committed:
Somme corrette senza promo

Wayde07 committed:
Definitivo

Compito finito con qualche incertezza

danielebarbaro commented on receipt/scontrino0.txt:
```diff
@@ -0,0 +1,17 @@
++ -------------------------------------------------- +
+	
+** -------------------------------------------------- **
+	Winston Wolf ha un credito residuo di 10.04
```
Perché tutte le soluzioni hanno una doppia riga (una giusta e una sbagliata) ?

Guarda riga 27 e 29 di cart.mjs

Wayde07 replied:
Questo purtroppo, come ho scritto nel commento all'interno del codice, non riesco a creare un metodo che richiami o un o l'altro in base alla presenza del promo code

danielebarbaro replied:
Va bene, ricordamelo in classe che lo vediamo insieme.

Wayde07 committed:
Definitivo2

Nome della cartella receipts e degli scontrini risolto.
Il fatto della doppia riga non riesco a capire come correggere e quindi le ho messe entrambe...
